incubator/hnc/pkg/reconcilers/hierarchy_config.go

```diff
@@ -101,24 +115,17 @@ func (r *HierarchyConfigReconciler) reconcile(ctx context.Context, log logr.Logg
 	origHC := inst.DeepCopy()
 	origNS := nsInst.DeepCopy()
 
-	// If either object exists but is being deleted, we won't update them when we're finished syncing
-	// (we should sync our internal data structure anyway just in case something's changed).  I'm not
-	// sure if this is the right thing to do but the kubebuilder boilerplate included this case
-	// explicitly.
-	update := true
-	if !inst.GetDeletionTimestamp().IsZero() || !nsInst.GetDeletionTimestamp().IsZero() {
-		log.Info("Singleton or namespace are being deleted; will not update")
-		update = false
+	// Set finalizers first. Remove them in the deleting() later if applicable.
+	inst.ObjectMeta.Finalizers = []string{api.MetaGroup}
+
+	// If the instance is being deleted, remove the finalizers accordingly.
+	if !inst.DeletionTimestamp.IsZero() {
+		r.updateFinalizers(ctx, log, inst, nsInst, hnsnms)
 	}
```

Comment:
Alternatively, `updateFinalizers` could contain all this logic? Nice simplification btw!

Reply:
Done. Put all above into `updateFinalizers()`.